### PR TITLE
Adds submenu to personalization category, addons listing page.

### PR DESCRIPTION
**Pull request by chuckharmston**
screen shot 2015-08-27 at 3 08 55 pm
screen shot 2015-08-27 at 3 09 16 pm

### CODE REVIEW

**Comment by chuckharmston** [comment]
Addon listing page is currently 500ing due to http://sentry.dmz.phx1.mozilla.com/marketplace-dev/marketplace-dev/group/28836/, but this should cover everything.

File: src/media/js/routes.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ define('routes',
     router.addRoutes([
         {'pattern': '^/(app.html|index.html)?$', 'view_name': 'homepage'},
         {'pattern': '^/addon/([^/<>"\']+)/?$', 'view_name': 'addon'},
+        {'pattern': '^/addons', 'view_name': 'addons'},
```
Comment (ngokevin) [inline src/media/js/routes.js]: add trailing slash

File: src/media/js/views/addons.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+define('views/addons',
+    ['core/capabilities', 'core/l10n', 'core/settings', 'utils_local'],
+    function(caps, l10n, settings, utils) {
+    'use strict';
+    var gettext = l10n.gettext;
+
+    return function(builder, args) {
+        builder.z('type', 'leaf');
+        builder.z('title', gettext('Loading...'));
```
Comment (ngokevin) [inline src/media/js/views/addons.js]: you can set a title here

File: src/templates/_macros/addon_tile.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{% macro addon_tile(addon) %}
+  <article class="addon">
+    <h1><a href="{{api('addon', [addon.slug])}}">{{ addon.name }}</a></h1>
```
Comment (ngokevin) [inline src/templates/_macros/addon_tile.html]: spaces in braces

File: src/media/js/views/addons.js (hunk 2 above)
Comment (chuckharmston) [inline src/media/js/views/addons.js]: :spaghetti:

File: src/templates/_macros/addon_tile.html (hunk 3 above)
Comment (ngokevin) [inline src/templates/_macros/addon_tile.html]: ```url('addon', [addon.slug])``` not api

File: src/templates/addon/list.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{% include '_macros/addon_tile.html' %}
+
+<section class="main">
+  <div class="subheader">
+    <h1>{{ _('Firefox OS Add-ons') }}</h1>
+  </div>
+</section>
+
+<section class="main full c">
+  {% defer (url=api('addon_list'), pluck='objects', as=addons) %}
```
Comment (ngokevin) [inline src/templates/addon/list.html]: do we need ```key='slug'```? forgot whether that's magically handled

File: src/templates/addon/list.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        <li>
+          {{ addon_tile(addon) }}
+        </li>
+      {% endfor %}
+    </ul>
+  {% placeholder %}
+    <p class="spinner padded alt"></p>
+  {% empty %}
+    <p class="no-results">
+      {{ _('No addons found.') }}
```
Comment (ngokevin) [inline src/templates/addon/list.html]: s/addons/Firefox OS add-ons/g

File: src/templates/addon/list.html (hunk 4 above)
Comment (chuckharmston) [inline src/templates/addon/list.html]: I forget, too. Was hoping to get this merged and iterate if it didn't work once the endpoint is working.

Comment (ngokevin) [inline src/templates/addon/list.html]: i'd add the key i think you need it.

also ```as="addon"```, not ```as=addons```

**Comment by ngokevin** [comment]
I feel like we can add some space between the header/submenu
screen shot 2015-08-27 at 3 16 51 pm

**Comment by chuckharmston** [comment]
:skull:

**Comment by ngokevin** [comment]
i know UX plans on iterating upon this, but lol:

screen shot 2015-08-27 at 3 16 25 pm

**Comment by ngokevin** [comment]
r+wc

**Comment by chuckharmston** [comment]
Thanks!

**Pull request merged**